### PR TITLE
feat(codecov): Add onboarding radio selector for GHA or CLI

### PR DESCRIPTION
**Pull request by calvin-codecov**
Closes https://linear.app/getsentry/issue/CCMRG-426/add-radio-selector-for-choosing-between-gh-actions-and-cli

Screenshot 2025-05-05 at 4 29 13 PM

### CODE REVIEW

File: static/app/views/codecov/tests/tests.tsx [hunk 1]
```diff
@@ -25,6 +26,7 @@ export default function TestsPage() {
         </PageFilterBar>
       </PageFiltersContainer>
       <Summaries />
+      <TestsOnboardingPage />
```
Comment (adrianviquez) [inline static/app/views/codecov/tests/tests.tsx:29]: Shouldn't this be living in the onboarding route?

File: static/app/views/codecov/tests/onboarding/onboardingStep.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const Header = styled('h5')`
+  font-size: ${p => p.theme.fontSizeExtraLarge};
+  color: ${p => p.theme.gray300};
+`;
+
+const Content = styled('div')`
+  /* flex-grow: 1; */
+`;
+
+export const OnboardingStep = {
```
Comment (adrianviquez) [inline static/app/views/codecov/tests/onboarding/onboardingStep.tsx:23]: What is this being used for?

File: static/app/views/codecov/tests/onboarding/onboarding.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import {space} from 'sentry/styles/space';
+
+type SetupOption = 'githubAction' | 'cli';
+
+export default function TestsOnboardingPage() {
+  const [selectedOption, setSelectedOption] = useState<SetupOption>('githubAction');
+
+  return (
+    <LayoutGap>
+      <p>Test Analytics Onboarding</p>
```
Comment (adrianviquez) [inline static/app/views/codecov/tests/onboarding/onboarding.tsx:15]: I forgot, we should be using the t() library for everything 👌

File: static/app/views/codecov/tests/onboarding.tsx [hunk 4]
```diff
@@ -1,13 +1,46 @@
...
+      <OnboardingContainer>
+        <IntroContainer>
+          <GetStartedHeader>{t('Get Started with Test Analytics')}</GetStartedHeader>
+          <TAValueText>
+            {t(
+              'Test Analytics offers data on test run times, failure rates, and identifies flaky tests to help decrease the risk of deployment failures and make it easier to ship new features quickly.'
+            )}
+          </TAValueText>
+        </IntroContainer>
+        <SelectOptionHeader>Select a setup option</SelectOptionHeader>
```
Comment (adrianviquez) [inline static/app/views/codecov/tests/onboarding.tsx:33]: t()

File: static/app/views/codecov/tests/onboarding.tsx [hunk 5]
```diff
@@ -1,13 +1,46 @@
...
 
+import RadioGroup from 'sentry/components/forms/controls/radioGroup';
+import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
-import TestPreOnboardingPage from 'sentry/views/codecov/tests/preOnboarding';
+
+type SetupOption = 'githubAction' | 'cli';
 
 export default function TestsOnboardingPage() {
+  const [searchParams, setSearchParams] = useSearchParams();
```
Comment (adrianviquez) [inline static/app/views/codecov/tests/onboarding.tsx:12]: I'm curious as to why we're using the url to handle state vs a local useState. Are we expecting to reach the /new route with a url param that could be any of these two? And what is the value of exposing the opt to the customer via the URL vs having it in an internal state?

File: static/app/views/codecov/tests/onboarding.spec.tsx [hunk 6]
```diff
@@ -1,18 +1,81 @@
...
+import TestsOnboardingPage from 'sentry/views/codecov/tests/onboarding';
 
-import TestAnalyticsOnboardingPage from 'sentry/views/codecov/tests/onboarding';
+describe('TestsOnboardingPage', () => {
+  it('renders with GitHub Actions selected by default', () => {
+    render(<TestsOnboardingPage />, {
+      initialRouterConfig: {
+        location: {
+          pathname: '/codecov/tests/new',
+          query: {opt: ''},
```
Comment (adrianviquez) [inline static/app/views/codecov/tests/onboarding.spec.tsx:11]: I think it'd be worth adding a test with query being empty/null; subtle but different is that the radio button will have the default value (gha) but the opt will be '' (or undefined if it's just `/new`), leaving the url and button out of sync. Sort of tied to my other comment of having the url be the state vs a useState hook

File: static/app/views/codecov/tests/onboarding.tsx (hunk 5 above)
Comment (calvin-codecov) [reply static/app/views/codecov/tests/onboarding.tsx:12]: Ahh, it was on this ticket https://linear.app/getsentry/issue/CCMRG-150/test-analytics-onboarding-create-radio-selection. Seems like we may have two similar/duplicate tickets for the radio selector that were created at different times. This is the other one I have included in the PR description https://linear.app/getsentry/issue/CCMRG-426/add-radio-selector-for-choosing-between-gh-actions-and-cli

Comment (adrianviquez) [reply static/app/views/codecov/tests/onboarding.tsx:12]: Ahh gotcha gotcha, ok sounds good! Yeap if this is expected to be part of the URL then this approach is good.

Should the url be populated with the default value if one isn't provided?

**Review by adrianviquez — APPROVED** [review]

File: static/app/views/codecov/tests/onboarding.tsx (hunk 5 above)
Comment (calvin-codecov) [reply static/app/views/codecov/tests/onboarding.tsx:12]: Do you see any benefits we would get with the default value being added? I was thinking of setting the searchParams on mount to add the query param by default but that forces a remount which I think may not be worth it if it already works when no default is specified.

Comment (adrianviquez) [reply static/app/views/codecov/tests/onboarding.tsx:12]: Mostly it being in sync with the app's state. When you click on another entry it syncs, which is nice, but I suppose if you default to an option it's okay as long as there's a default for blank.
